src/dmd/dtoh.d

```diff
@@ -180,6 +180,21 @@ struct _d_dynamicArray final
 `);
     }
 
+    // prevent trailing newlines
+    foreach_reverse (i, ch; buf)
```

Comment:
Maybe this is overkill logic. Checking for `['\n', '\n']` should be enough, but if any node introduces double newlines, this is required.

Reply:
Check for repeated newlines (either `\r\n`  or `\n`) should be sufficient

Reply:
You mean, specifically only twice or more than one? I don't see any case where dtoh generates `\n\n\n`, so maybe the twice case is sufficient.

Reply:
More than one. I don't expect more than two but just to be sure.